### PR TITLE
Apparmor: fix various denials

### PR DESCRIPTION
**Pull request by Vincent43**
Fixes following erros:

wine:
```
AVC apparmor="DENIED" operation="unlink" profile="firejail-default" name="/run/firejail/profile/11526" pid=11533 comm="wine" requested_mask="d" denied_mask="d" fsuid=1000 ouid=0
AVC apparmor="DENIED" operation="unlink" profile="firejail-default" name="/run/firejail/profile/5807" pid=11533 comm="wine" requested_mask="d" denied_mask="d" fsuid=1000 ouid=0
AVC apparmor="DENIED" operation="unlink" profile="firejail-default" name="/run/firejail/profile/2017" pid=11533 comm="wine" requested_mask="d"

```
cups:
```
AVC apparmor="DENIED" operation="connect" profile="firejail-default" name="/run/cups/cups.sock" pid=11682 comm="lpr" requested_mask="w" denied_mask="w" fsuid=1000 ouid=0
AVC apparmor="DENIED" operation="connect" profile="firejail-default" name="/run/cups/cups.sock" pid=11682 comm="lpr" requested_mask="w" denied_mask="w" fsuid=1000 ouid=0
AVC apparmor="DENIED" operation="connect" profile="firejail-default" name="/run/cups/cups.sock" pid=11682 comm="lpr" requested_mask="w" denied_mask="w" fsuid=1000 ouid=0
AVC apparmor="DENIED" operation="connect" profile="firejail-default" name="/run/cups/cups.sock" pid=11682 comm="lpr" requested_mask="w" denied_mask="w" fsuid=1000 ouid=0
AVC apparmor="DENIED" operation="connect" profile="firejail-default" name="/run/cups/cups.sock" pid=11682 comm="lpr" requested_mask="w" denied_mask="w" fsuid=1000 ouid=0

```
chromium:
```
AVC apparmor="DENIED" operation="open" profile="firejail-default" name="/proc/8/mem" pid=7858 comm="chromium" requested_mask="r" denied_mask="r" fsuid=1000 ouid=1000
AVC apparmor="DENIED" operation="open" profile="firejail-default" name="/proc/8/oom_score_adj" pid=7858 comm="chromium" requested_mask="wc" denied_mask="wc" fsuid=1000 ouid=1000
AVC apparmor="DENIED" operation="open" profile="firejail-default" name="/proc/11/mem" pid=7861 comm="chromium" requested_mask="r" denied_mask="r" fsuid=1000 ouid=1000
AVC apparmor="DENIED" operation="open" profile="firejail-default" name="/proc/sys/kernel/yama/ptrace_scope" pid=7861 comm="chromium" requested_mask="r" denied_mask="r" fsuid=1000 ouid=0
AVC apparmor="DENIED" operation="ptrace" profile="firejail-default" pid=7858 comm="chromium" requested_mask="trace" denied_mask="trace" peer="firejail-default"
AVC apparmor="DENIED" operation="ptrace" profile="firejail-default" pid=7858 comm="chromium" requested_mask="tracedby" denied_mask="tracedby" peer="firejail-default"
AVC apparmor="DENIED" operation="ptrace" profile="firejail-default" pid=7858 comm="TaskSchedulerBa" requested_mask="trace" denied_mask="trace" peer="firejail-default"
AVC apparmor="DENIED" operation="ptrace" profile="firejail-default" pid=7858 comm="TaskSchedulerBa" requested_mask="tracedby" denied_mask="tracedby" peer="firejail-default"
AVC apparmor="DENIED" operation="open" profile="firejail-default" name="/proc/46/mem" pid=7897 comm="chromium" requested_mask="r" denied_mask="r" fsuid=1000 ouid=1000
AVC apparmor="DENIED" operation="open" profile="firejail-default" name="/proc/46/oom_score_adj" pid=7897 comm="chromium" requested_mask="wc" denied_mask="wc" fsuid=1000 ouid=1000
AVC apparmor="DENIED" operation="open" profile="firejail-default" name="/proc/sys/kernel/yama/ptrace_scope" pid=7897 comm="chromium" requested_mask="r" denied_mask="r" fsuid=1000 ouid=0
AVC apparmor="DENIED" operation="open" profile="firejail-default" name="/proc/58/oom_score_adj" pid=7910 comm="chrome-sandbox" requested_mask="w" denied_mask="w" fsuid=0 ouid=0
AVC apparmor="DENIED" operation="open" profile="firejail-default" name="/proc/58/oom_adj" pid=7910 comm="chrome-sandbox" requested_mask="w"
```

### CODE REVIEW

**Comment by smitsohu** [comment]
Hi @Vincent43, thanks for the pull request. Can I ask you to elaborate a bit on the underlying issue(s)? If the issue is a polluted syslog, maybe an alternative to allowing these paths could be to explicitly `deny` them.

File: etc/firejail-default [hunk 1]
```diff
@@ -134,6 +145,11 @@ owner /run/firejail/mnt/oroot/{run,dev}/shm/** rmwk,
 /run/firejail/mnt/oroot/opt/** r,
 /run/firejail/mnt/oroot/opt/** ix,
 
+##########
+# Allow acces to cups printing socket
+##########
+/run/cups/cups.sock w,
```
Comment (smitsohu) [inline etc/firejail-default]: Thanks, this probably fixes #1015 :+1:

File: etc/firejail-default [hunk 2]
```diff
@@ -95,15 +99,22 @@ owner /run/firejail/mnt/oroot/{run,dev}/shm/** rmwk,
...
+owner /proc/@{PID}/oom_adj w,
 /proc/@{PID}/oom_score_adj r,
+owner /proc/@{PID}/oom_score_adj w,
 /proc/@{PID}/auxv r,
 /proc/@{PID}/net/dev r,
 /proc/@{PID}/loginuid r,
 /proc/@{PID}/environ r,
 
+# Needed for chromium
+ptrace (trace tracedby),
```
Comment (smitsohu) [inline etc/firejail-default]: afaik only the chromium crash handler needs this.

Comment (Vincent43) [reply etc/firejail-default]: Should we allow it or not? Maybe leave it commented out it by default same as dbus?

**Comment by Vincent43** [comment]
As for wine I have no idea what it's doing. No breakages, I didn't observe it with any other app.

As for chromium this interferes with their sandbox, i.e. in `chrome://sandbox` Yama LSM Enforcing is disabled. Similar lines are in upstream apparmor profile: https://gitlab.com/apparmor/apparmor-profiles/blob/master/ubuntu/18.04/usr.bin.chromium-browser#L33

**Comment by smitsohu** [comment]
@Vincent43 Thanks for the update.

@netblue30 I am a bit concerned about the growing number of additions to the /proc section of the profile. Already the following lines have found their way into the profile based on audit logs, and not because apparmor broke something (or so it seemed at least):
```
/proc/@{PID}/auxv r,
/proc/@{PID}/net/dev r,
/proc/@{PID}/loginuid r,
/proc/@{PID}/environ r,
```
which makes me wonder if section headline and firejail man page are actually still telling the truth:

> Mask /proc and /sys information leakage. The configuration here is barely
enough to run "top" or "ps aux".

In order to address the issue of noisy syslogs (and it can be a lot of noise), it might make sense to silence audit messages with `deny` in some cases, but on the downside this generally makes debugging more difficult.

Another question is how we should handle ptrace. If we want to delegate all ptrace mediation to firejail, we need to unconditionally allow it in the apparmor profile: `ptrace (read readby trace tracedby)`
Personally I would find a commented ptrace, as suggested by @Vincent43, quite appealing...

Opinions?

**Comment by Vincent43** [comment]
Yeah, there is a problem with "one size fits all" approach. Especially when building whitelist from hundreds of files under /proc and /sys. I wonder if we should switch to blacklist instead - disable access to known sensitive files and allow the rest?

**Comment by Vincent43** [comment]
Another question is if apparmor should deal at all with `/proc` and `/sys`. Maybe leave it for firejail the same as other dirs, caps, mounts, etc.  Apparmor can focus on dbus, ptrace, signals, execution, etc.

**Comment by Vincent43** [comment]
@netblue30 should I create PR which switch `/proc` and `/sys` handling to blacklist and blacklist same things which firejail do?

**Pull request merged**